storage/src/vespa/storage/persistence/persistencethread.cpp

```diff
@@ -205,53 +203,66 @@ PersistenceThread::handleRemove(api::RemoveCommand& cmd, MessageTracker::UP trac
         return trackerUP;
     }
 
+    spi::Bucket bucket = getBucket(cmd.getDocumentId(), cmd.getBucket());
     if (_sequencedExecutor == nullptr) {
-        spi::RemoveResult response = _spi.removeIfFound(getBucket(cmd.getDocumentId(), cmd.getBucket()),
-                                                        spi::Timestamp(cmd.getTimestamp()), cmd.getDocumentId(),
-                                                        tracker.context());
+        spi::RemoveResult response = _spi.removeIfFound(bucket, spi::Timestamp(cmd.getTimestamp()), cmd.getDocumentId(),tracker.context());
         if (tracker.checkForError(response)) {
             tracker.setReply(std::make_shared<api::RemoveReply>(cmd, response.wasFound() ? cmd.getTimestamp() : 0));
         }
         if (!response.wasFound()) {
             metrics.notFound.inc();
         }
     } else {
-        _spi.removeIfFoundAsync(getBucket(cmd.getDocumentId(), cmd.getBucket()),
-                         spi::Timestamp(cmd.getTimestamp()), cmd.getDocumentId(), tracker.context(),
-                         std::make_unique<ResultTaskOperationDone>(*_sequencedExecutor, cmd.getBucketId(),
-                               makeResultTask([&metrics, &cmd, tracker = std::move(trackerUP)](spi::Result::UP responseUP) {
-                                   const spi::RemoveResult & response = dynamic_cast<const spi::RemoveResult &>(*responseUP);
-                                   if (tracker->checkForError(response)) {
-                                       tracker->setReply(std::make_shared<api::RemoveReply>(cmd, response.wasFound() ? cmd.getTimestamp() : 0));
-                                   }
-                                   if (!response.wasFound()) {
-                                       metrics.notFound.inc();
-                                   }
-                                   tracker->sendReply();
-                               })));
+        auto task = makeResultTask([&metrics, &cmd, tracker = std::move(trackerUP)](spi::Result::UP responseUP) {
```

Comment:
Consider adding a comment stating why it's safe to refer to `cmd` as a reference capture to the argument given to the function itself, since the lifetime semantics aren't necessarily obvious.

Reply:
Fixed